src/frontend/src/icp/workers/icrc-wallet.worker.ts

```diff
@@ -25,13 +25,18 @@ const getTransactions = ({
 }: SchedulerJobParams<PostMessageDataRequestIcrc>): Promise<IcrcIndexNgGetTransactions> => {
 	assertNonNullish(data, 'No data - indexCanisterId - provided to fetch transactions.');
 
+	// TODO(OISY-296): This is not clean. If the index canister ID is not provided we should not even land here.
+	const { indexCanisterId } = data;
+	assertNonNullish(indexCanisterId);
+
 	return getTransactionsApi({
 		identity,
 		certified,
 		owner: identity.getPrincipal(),
 		// We query tip to discover the new transactions
 		start: undefined,
-		...data
+		...data,
+		indexCanisterId
```

Comment:
if you provide only `...data` , it still gives an error because it does not understand that `indexCanisterId` is not nullish?

Reply:
Exactly. Even if I `assertNonNullish(data.indexCanisterId);` it still do not interfere it therefore explicitely providing the nonnullish `indexCanisterId`.